src/components/02-organisms/Attachments/index.js

```diff
@@ -172,426 +175,434 @@ export const Input = styled.input`
-        ) {
-          let atts = props.payload.attachments;
-          atts[idx].nimi = selectedAttachment.nimi;
-          setAttachments(atts);
-          props.onUpdate(props.payload, {
-            attachments: atts
-          });
-        }
-        return null;
-      });
-    } else setNameMissing(true);
-  };
-
-  const setAttachment = e => {
-    setFileError(false);
-    // setFileAdded("");
-
-    if (e.target.files.length === 0) return;
-    console.log("File selected");
-
-    const type = e.target.files[0].name
-      .split(".")
-      .pop()
-      .toLowerCase();
-
-    // Rajoitetaan max kooksi 25MB ja vain pdf, word, excel, jpeg ja gif on sallittuja
-    if (
-      e.target.files[0].size <= 26214400 &&
-      [
-        "pdf",
-        "doc",
-        "txt",
-        "docx",
-        "xls",
-        "xlsx",
-        "xlsm",
-        "jpg",
-        "jpeg",
-        "jpe",
-        "jfif",
-        "gif"
-      ].includes(type)
-    ) {
-      let liite = {};
-      liite.tiedostoId = e.target.files[0].name + "-" + Math.random();
-      liite.kieli = "fi";
-      liite.tyyppi = type;
-      liite.nimi = e.target.files[0].name.split(".")[0].toLowerCase();
-      liite.tiedosto = new Blob([e.target.files[0]]);
-      liite.koko = e.target.files[0].size;
-      liite.removed = false;
-      liite.salainen = false;
-      liite.paikka = props.placement;
-      liite.new = true;
-
-      let atts = props.payload.attachments;
-      atts.push(liite);
-      setAttachments(atts);
-      setSelectedAttachment(liite);
+    const {
+      intl: { formatMessage }
+    } = props;
 
-      openNameModal();
-    } else return setFileError(true);
-  };
+    const openNameModal = () => {
+      setNameMissing(false);
+      setIsNameModalOpen(true);
+    };
 
-  const cancelAttachment = () => {
-    closeNameModal();
-    let atts = attachments.pop(); // take last out
-    setAttachments(atts);
-  };
+    const closeNameModal = () => {
+      setIsNameModalOpen(false);
+    };
 
-  const removeAttachment = (e, tiedostoId, uuid) => {
-    e.preventDefault();
-    setFileError(false);
+    if (props.payload && props.payload.attachments) {
+      useEffect(() => {
+        setAttachments(props.payload.attachments);
+      }, [props.payload.attachments]);
+    }
 
-    attachments.map((liite, idx) => {
-      if (
-        (tiedostoId && liite.tiedostoId === tiedostoId) ||
-        (uuid && liite.uuid === uuid)
-      ) {
-        let atts = [...props.payload.attachments];
-        atts[idx].removed = true;
-        setAttachments(atts);
-        props.onUpdate(props.payload, {
-          attachments: atts
+    // Adds attachment into attachment list
+    const addAttachment = () => {
+      if (selectedAttachment.nimi) {
+        setNameMissing(false);
+        closeNameModal();
+        // Find correct attachment and set name
+        props.payload.attachments.map((liite, idx) => {
```

Comment:
This loop (or actually mapping of objs) can be replaced with some kind of find function?

Reply:
Yeah, it could. But need to continue if not that bad this way either.